### ISSUE TITLE
Enable configfs support for USB Gadget

### ISSUE THREAD

**Issue by zerotri** [NextThingCo/CHIP-linux#19]
Was given the following config in a support ticket
[config.txt](https://github.com/NextThingCo/CHIP-linux/files/335600/config.txt)

**Comment by hadess** [comment]
Yes. The current USB gadget setup is nice (Ethernet + serial as used in 4.4), but if one wants to add support for more gadget functionality, they'd need to disable that functionality, and switch kernels.

I'd be interested in implementing either PTP or storage in a CHIP, so I can plug it in a computer and add media files for it.
